core/src/test/scala/integration/kafka/api/AbstractConsumerTest.scala

```diff
@@ -94,12 +94,18 @@ abstract class AbstractConsumerTest extends BaseRequestTest {
   def awaitNonEmptyRecords[K, V](consumer: Consumer[K, V],
                                  partition: TopicPartition,
                                  pollTimeoutMs: Long = 100): ConsumerRecords[K, V] = {
+    var result: ConsumerRecords[K, V] = null
+
     TestUtils.pollRecordsUntilTrue(consumer, (polledRecords: ConsumerRecords[K, V]) => {
-      if (polledRecords.records(partition).asScala.nonEmpty)
-        return polledRecords
-      false
+      if (polledRecords.records(partition).asScala.nonEmpty) {
+        result = polledRecords
+        true
+      } else {
+        false
+      }
```

Comment:
This part can be simplified.
```suggestion
      val hasRecords = !polledRecords.records(partition).isEmpty
      if (hasRecords) result = polledRecords
      hasRecords
```

Reply:
Great suggestion! Quick question about style consistency. Should we use `.isEmpty` directly or keep `.asScala.nonEmpty` for consistency with lines 251-252? Happy to update either way!

Reply:
We can directly use `.isEmpty` to avoid unnecessary conversion from Java collections to Scala collections.

Reply:
Updated, thanks for the feedback! PTAL~